src/App.test.js

```diff
@@ -1,8 +1,44 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
-
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
-});
+// Imports React into our test file.
+import React from 'react'
+
+// Imports Enzyme testing and deconstructs Shallow into our test file.
+import Enzyme, { shallow } from 'enzyme'
+
+// Imports Adapter utilizing the latest react version into our test file so we can run a testing render on any component we may need.
+import Adapter from 'enzyme-adapter-react-16'
+
+// Imports in the component we are going to be testing.
+import App from './App'
+import NotFound from './Pages/NotFound/NotFound'
+
+//Allows us to utilize the adapter we import in earlier, allowing us to call and render a component.
+Enzyme.configure({adapter: new Adapter()})
+
+describe("When App renders", () => {
+  it("displays a heading", () => {
+    const renderedApp = shallow(<App />)
+
+    const renderedHeader = renderedApp.find('Header')
+    const renderedFooter = renderedApp.find('Footer')
+    
+    expect(renderedHeader.length).toEqual(1)
+    expect(renderedFooter.length).toEqual(1)
+  })
+
+  it("provides a route of '/' to the home component", () => {
+    const renderedApp = shallow(<App />)
+
+    const renderedHomeRoute = renderedApp.find("[path='/']")
+    
+    expect(renderedHomeRoute.length).toEqual(1)
+})
+
+it("displays a page not found", () => {
```

Comment:
This test should be in a NotFound.test.js file

Reply:
we made a NotFound.test.js file and we have our testing there